ckan/tests/logic/action/test_get.py

```diff
@@ -852,6 +852,17 @@ def test_search(self):
         eq(search_result['results'][0]['title'], 'Rivers')
         eq(search_result['count'], 1)
 
+    def test_search_fl(self):
+        factories.Dataset(title='Rivers', name='test_ri')
+        factories.Dataset(title='Lakes')
+
+        search_result = helpers.call_action('package_search', q='rivers', fl=['title', 'name'])
+
+        eq(search_result['results'][0]['title'], 'Rivers')
+        eq(search_result['results'][0]['name'], 'test_ri')
+        eq(len(search_result['results'][0]), 2)
+        eq(search_result['count'], 1)
```

Comment:
you can replace these four lines with:
```python
eq(search_result['results'], [{'title': 'Rivers', 'name': 'test_ri'}])
```
for less code and a better error message when the test fails (the error will include the complete results list)